src/python/WMComponent/DBS3Buffer/DBSUploadPoller.py

```diff
@@ -110,14 +110,29 @@ def uploadWorker(workInput, results, dbsUrl, gzipEncoding=False):
                 logging.warning(msg)
                 results.put({'name': name, 'success': "error", 'error': msg})
             else:
-                msg = "Error trying to process block %s through DBS. Error: %s" % (name, exString)
+                reason = parseDBSException(exString)
+                msg = "Error trying to process block %s through DBS. Error: %s" % (name, reason)
                 logging.exception(msg)
                 logging.debug("block info: %s \n", block)
                 results.put({'name': name, 'success': "error", 'error': msg})
 
     return
 
 
+def parseDBSException(exc):
+    """
+    parse DBS Go-based server exception
+    :param exc: exception message
+    :return: either (parsed) concise exception message or original exception
+    """
+    try:
+        data = json.loads(exc)
+        # dbs2go always return a list
+        return data[0]['error']['reason']
+    except:
+        return exc
```

Comment:
I doubt it, the `reason` part of DBS server response is a string. And, wrapping exception into string is wrong since it is up to upstream code to decide what to do with exception. Unless you provide real reason for casting I'll not do it.